pkg/grpc/api.go

```diff
@@ -1580,14 +1580,13 @@ func (a *api) GetConfigurationAlpha1(ctx context.Context, in *runtimev1pb.GetCon
 		return &runtimev1pb.GetConfigurationResponse{}, err
 	}
 
-	cachedItems := make([]*commonv1pb.ConfigurationItem, 0)
-	for _, v := range getResponse.Items {
-		cachedItems = append(cachedItems, &commonv1pb.ConfigurationItem{
-			Key:      v.Key,
+	cachedItems := make(map[string]*commonv1pb.ConfigurationItem, len(getResponse.Items))
```

Comment:
is there a reason for this to be a pointer to ConfigItem?

Reply:
@mukundansundar this is not my PR. This is from @beiwei30 - I don't know. I just dealt with the messy merge conflicts, missing tests and changes, and other broken things. If maintainers don't want it to be a pointer, please go ahead and fix up this PR here.

Reply:
@dapr/maintainers-dapr any thoughts on avoiding pointers unless necessary?